packages/web/pages/stake/index.tsx

```diff
@@ -268,6 +274,41 @@ export const Staking: React.FC = observer(() => {
 
   const showStakeLearnMore = !isWalletConnected || isNewUser;
 
+  const { unbondingBalances } = unbondingDelegationsQuery;
+  const unbondingInProcess = unbondingBalances.length > 0;
+
+  function groupByCompletionTime(
+    array: Array<{
+      validatorAddress: string;
+      entries: { completionTime: string; balance: CoinPretty }[];
+    }>
+  ): { completionTime: string; balance: CoinPretty }[] {
+    const groupedObjects: Record<string, CoinPretty> = {};
```

Comment:
Personally, I'd prefer a `Map` over using an object, with `Object.*` utils.

Also, perhaps it's possible to use flatMap or reduce over this "array" object? Instead of nested for loops. From experience, the only use case for `for` loops is when you want to save some iterations via `continue`. Otherwise functional programming is preferred.

Also, what is the content of completion time. Is it an ISO string? IMO proper Date or dayjs duration objects offer more type safety then string representations. These object ideally should be created at the query store later and consumed by the view.

Reply:
I can get rid of the nested loop. Good catch. 

Map and Object aren't going to have any functional difference here, since I am just storing strings and values.

`completionTime` is just whatever string is returned by the chain. I don't want to change the type since all I'm doing is a groupBy.